File: Java/Ohne Polymorphismus/OhnePolymorphismus.java
import java.util.ArrayList;

public class OhnePolymorphismus {
    public static void main(String[] args) {
        Haustier h1 = new Haustier(HaustierArt.HUND, "Bello");
        Haustier h2 = new Haustier(HaustierArt.KATZE, "Mietze");
        Haustier h3 = new Haustier(HaustierArt.WELLENSITTICH, "Hansi");
        Haustier h4 = new Haustier(HaustierArt.HUND, "Lopi");

        ArrayList<Haustier> haustiere = new ArrayList<>();

        haustiere.add(h1);
        haustiere.add(h2);
        haustiere.add(h3);
        haustiere.add(h4);

        for (Haustier tier : haustiere){
            tier.schreibeName();
            tier.fressen();
            tier.freuen();
            tier.lautGeben();
            System.out.println();
        }
    }
}

File: C#/Ohne Polymorphismus/OhnePolymorphismus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


class OhnePolymorphismus
{
    static void Main(string[] args)
    {
        Haustier h1 = new Haustier(HaustierArt.HUND, "Bello");
        Haustier h2 = new Haustier(HaustierArt.KATZE, "Mietze");
        Haustier h3 = new Haustier(HaustierArt.WELLENSITTICH, "Hansi");
        Haustier h4 = new Haustier(HaustierArt.HUND, "Lopi");

        List<Haustier> haustiere =
            new List<Haustier>() { h1, h2, h3, h4 };

        foreach (Haustier tier in haustiere)
        {
            tier.SchreibeName();
            tier.Fressen();
            tier.Freuen();
            tier.LautGeben();
            System.Console.WriteLine();
        }

        Console.ReadKey();
    }
}

